### PR TITLE
Tests for Epoch Reorgs And Balance Deviations

### PR DESCRIPTION
**Pull request by rauljordan**
This is part of #2229 

---

# Description

**Write why you are making the changes in this pull request**

We have identified that balances across nodes differ after a node goes through an epoch reorg. This should be easy to verify through unit tests to discover if there is any sort of inconsistency between 2 different nodes.

**Write a summary of the changes you are making**

This PR implements unit tests for ensuring balances match after an epoch reorg between two nodes.

### CODE REVIEW

File: beacon-chain/blockchain/block_processing.go [hunk 1]
```diff
@@ -129,6 +129,7 @@ func (c *ChainService) ApplyBlockStateTransition(
 	// Check for skipped slots.
 	numSkippedSlots := 0
 	for beaconState.Slot < block.Slot-1 {
+		fmt.Printf("SKIP SLOT: %d\n", beaconState.Slot-params.BeaconConfig().GenesisSlot)
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/block_processing.go]: remove or use proper logs

File: beacon-chain/blockchain/fork_choice.go [hunk 2]
```diff
@@ -134,10 +134,15 @@ func (c *ChainService) ApplyForkChoiceRule(
...
 		return fmt.Errorf("could not retrieve attestation target: %v", err)
 	}
+	for idx, target := range attestationTargets {
+		fmt.Printf("Index: %d, target slot: %d\n", idx, target.Slot-params.BeaconConfig().GenesisSlot)
+	}
 	justifiedHead, err := c.beaconDB.JustifiedBlock()
 	if err != nil {
 		return fmt.Errorf("could not retrieve justified head: %v", err)
 	}
+	fmt.Printf("Justified head: %v\n", justifiedHead.Slot-params.BeaconConfig().GenesisSlot)
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/fork_choice.go]: remove or use proper logs

File: beacon-chain/blockchain/fork_choice.go [hunk 3]
```diff
@@ -134,10 +134,15 @@ func (c *ChainService) ApplyForkChoiceRule(
...
+	fmt.Printf("Justified state: %v\n", justifiedState.Slot-params.BeaconConfig().GenesisSlot)
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/fork_choice.go]: remove or use proper logs

File: beacon-chain/blockchain/fork_choice.go [hunk 4]
```diff
@@ -134,10 +134,15 @@ func (c *ChainService) ApplyForkChoiceRule(
 	if err != nil {
 		return fmt.Errorf("could not retrieve attestation target: %v", err)
 	}
+	for idx, target := range attestationTargets {
+		fmt.Printf("Index: %d, target slot: %d\n", idx, target.Slot-params.BeaconConfig().GenesisSlot)
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/fork_choice.go]: remove or use proper logs

File: beacon-chain/blockchain/fork_choice_reorg_test.go [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+
+type mockAttestationManager struct {
+	targets map[uint64]*pb.BeaconBlock
+	beaconDB *db.BeaconDB
+}
+
+func (m *mockAttestationManager) LatestAttestationTarget(ctx context.Context, validatorIndex uint64) (*pb.BeaconBlock, error) {
+	return m.targets[validatorIndex], nil
+}
+func (m *mockAttestationManager) UpdateLatestAttestation(ctx context.Context, attestation *pb.Attestation) error {
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/fork_choice_reorg_test.go]: Does this need to be an implemented mock or can you use the real service?

**Comment by rauljordan** [comment]
Closing in favor of the tests added in #2413

**Pull request closed without merging**